pkg/docker/docker.go

```diff
@@ -592,14 +594,23 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) (err error) {
 		config.AttachStdout = true
 	}
 
-	glog.V(2).Infof("Creating container using config: %+v", config)
 	ccopts := docker.CreateContainerOptions{Name: "", Config: &config}
+	ccopts.HostConfig = &docker.HostConfig{}
 	if opts.TargetImage {
-		ccopts.HostConfig = &docker.HostConfig{PublishAllPorts: true, NetworkMode: opts.NetworkMode}
+		ccopts.HostConfig.PublishAllPorts = true
+		ccopts.HostConfig.NetworkMode = opts.NetworkMode
 	} else if opts.NetworkMode != "" {
-		ccopts.HostConfig = &docker.HostConfig{NetworkMode: opts.NetworkMode}
+		ccopts.HostConfig.NetworkMode = opts.NetworkMode
 	}
 
+	if opts.CGroupLimits != nil {
+		ccopts.HostConfig.Memory = opts.CGroupLimits.MemoryLimitBytes
+		ccopts.HostConfig.MemorySwap = opts.CGroupLimits.MemorySwap
+		ccopts.HostConfig.CPUShares = opts.CGroupLimits.CPUShares
+		ccopts.HostConfig.CPUQuota = opts.CGroupLimits.CPUQuota
+		ccopts.HostConfig.CPUPeriod = opts.CGroupLimits.CPUPeriod
+	}
+	glog.V(2).Infof("Creating container using config: %s, %#v, %#v", ccopts.Name, ccopts.Config, ccopts.HostConfig)
```

Comment:
Isn't `%+v` enough?

I've used this in https://github.com/openshift/origin/pull/6715 and it is pretty verbose:

```go
Infof("Creating container with options {Name:%q Config:%+v HostConfig:%+v} ...", createOpts.Name, createOpts.Config, createOpts.HostConfig)
```

Example log output:

```
I0125 16:30:36.987207    1039 dockerutil.go:144] Creating container with options
{Name:"manual_testing_post_commit_hook" Config:&{Hostname: Domain name: User:
Memory:0 MemorySwap:0 MemoryReservation:0 KernelMemory:0 CPUShares:0 CPUSet:
AttachStdin:false AttachStdout:false AttachStderr:false P ortSpecs:[]
ExposedPorts:map[] StopSignal: Tty:false OpenStdin:false StdinOnce:false Env:[]
Cmd:[echo "Hello world!"] DNS:[] Image:c91ab321f208 Volumes:map[] VolumeDriver:
VolumesFrom: WorkingDir: MacAddress: Entrypoint:[sh -c] NetworkDisabled:false
SecurityOpts :[] OnBuild:[] Mounts:[] Labels:map[]} HostConfig:&{Binds:[]
CapAdd:[] CapDrop:[] GroupAdd:[] ContainerIDFile: LxcConf:[] Privileged:false
PortBin dings:map[] Links:[] PublishAllPorts:false DNS:[] DNSOptions:[]
DNSSearch:[] ExtraHosts:[] VolumesFrom:[] NetworkMode: IpcMode: PidMode:
UTSMode:  RestartPolicy:{Name: MaximumRetryCount:0} Devices:[] LogConfig:{Type:
Config:map[]} ReadonlyRootfs:false SecurityOpt:[] CgroupParent: Memory:0 Mem
orySwap:0 MemorySwappiness:0 OOMKillDisable:false CPUShares:1024 CPUSet:
CPUSetCPUs: CPUSetMEMs: CPUQuota:-1 CPUPeriod:100000 BlkioWeight:0 Ulimit s:[]
VolumeDriver:}} ...
```